src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.Globalization.Utf8.cs

```diff
@@ -0,0 +1,73 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    public static partial class MemoryExtensions
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static bool EqualsOrdinalIgnoreCaseUtf8(this ReadOnlySpan<byte> span, ReadOnlySpan<byte> value)
+        {
+            if (span.Length != value.Length)
```

Comment:
This shortcut is incorrect. In UTF-8, it is possible for two spans of different byte length to compare as equal under an OrdinalIgnoreCase comparison.

Reply:
Fixed.